RDF.js

```diff
@@ -428,71 +442,79 @@ function detectType(newItem, node, ret) {
 				t.so = 'artwork'; break;
 			case 'datacatalog':
 			case 'dataset':
-				t.so = 'journalArticle'; break;  //until dataset gets implemented
+				t.so = 'journalArticle'; break; // until dataset gets implemented
 			
 			// specials cases
 			case "article":
-				// choose between journal, newspaper, and magazine articles
-				// use of container = (not container ==) is intentional
-				if (container = getNodeByType(isPartOf,
-						[n.bib+"Journal", n.bibo+"Journal"])) {
+			// choose between journal, newspaper, and magazine articles
+				container = getNodeByType(isPartOf, [n.bib + "Journal", n.bibo + "Journal"]);
+				if (container) {
 					t.bib = "journalArticle";
-				} else if (container = getNodeByType(isPartOf,
-						[n.bib+"Periodical", n.bibo+"Periodical"])) {
+					break;
+				}
+				container = getNodeByType(isPartOf, [n.bib + "Periodical", n.bibo + "Periodical"]);
+				if (container) {
 					t.bib = "magazineArticle";
-				} else if (container = getNodeByType(isPartOf,
-						[n.bib+"Newspaper", n.bibo+"Newspaper"])) {
+					break;
+				}
+				container = getNodeByType(isPartOf, [n.bib + "Newspaper", n.bibo + "Newspaper"]);
+				if (container) {
 					t.bib = "newspaperArticle";
-				} else if (pref==n.so) {
-					if (container = getNodeByType(isPartOf,
-						[n.so+"PublicationIssue", n.so+"PublicationVolume"])) {
-							t.so = "journalArticle";
-					} else {
+					break;
```

Comment:
I replaced the old if-elseif-...else-structure with several `break`s here in this `case`.

Reply:
:heavy_check_mark: